gengokit/template/NAME-service/NAME-server/server_main.gotemplate

```diff
@@ -71,13 +71,16 @@ func main() {
 	{{end}}
 	)
 
-	// Inject Endpoint middlewares. See ../middlewares/endpoints.go
+	// Wrap Endpoints with middlewares. See ../middlewares/endpoints.go
 	endpoints := svc.Endpoints{
 	{{range $i := .Service.Methods -}}
-		{{$i.Name}}Endpoint:    middlewares.InjectEndpointMiddlewares({{ToLower $i.Name}}Endpoint),
+		{{$i.Name}}Endpoint:    middlewares.WrapEndpoint({{ToLower $i.Name}}Endpoint),
 	{{end}}
 	}
 
+	// Wrap selected Endpoints with middlewares. See ../middlewares/endpoints.go
+	endpoints = middlewares.WrapEndpoints(endpoints)
```

Comment:
I love this solution, pass the `svc.Endpoints` struct. I did not think of this at all.